taxi/views.py

```diff
@@ -104,16 +168,17 @@ class DriverLicenseUpdateView(LoginRequiredMixin, generic.UpdateView):
 
 class DriverDeleteView(LoginRequiredMixin, generic.DeleteView):
     model = Driver
-    success_url = reverse_lazy("")
+    success_url = reverse_lazy("taxi:driver-list")
 
 
 @login_required
 def toggle_assign_to_car(request, pk):
-    driver = Driver.objects.get(id=request.user.id)
-    if (
-        Car.objects.get(id=pk) in driver.cars.all()
-    ):  # probably could check if car exists
-        driver.cars.remove(pk)
+    car = get_object_or_404(Car, id=pk)
+    driver = get_object_or_404(Driver, id=request.user.id)
```

Comment:
The `toggle_assign_to_car` function retrieves the driver based on the logged-in user's ID. This assumes that the logged-in user is always the driver being modified, which might not be the intended behavior if an admin or another user type can perform this action. Consider verifying that the user has the appropriate permissions or explicitly passing the driver ID if needed.